File: openapi-generator/cpp-tizen/src/HosLogsSummaryResponseDrivers.cpp
#include <map>
#include <cstdlib>
#include <glib-object.h>
#include <json-glib/json-glib.h>
#include "Helpers.h"


#include "HosLogsSummaryResponse_drivers.h"

using namespace std;
using namespace Tizen::ArtikCloud;

HosLogsSummaryResponse_drivers::HosLogsSummaryResponse_drivers()
{
	//__init();
}

HosLogsSummaryResponse_drivers::~HosLogsSummaryResponse_drivers()
{
	//__cleanup();
}

void
HosLogsSummaryResponse_drivers::__init()
{
	//timeUntilBreak = long(0);
	//vehicleName = std::string();
	//drivingInViolationToday = long(0);
	//driverId = long(0);
	//cycleRemaining = long(0);
	//driverName = std::string();
	//dutyStatus = std::string();
	//cycleTomorrow = long(0);
	//shiftDriveRemaining = long(0);
	//timeInCurrentStatus = long(0);
	//drivingInViolationCycle = long(0);
	//shiftRemaining = long(0);
}

void
HosLogsSummaryResponse_drivers::__cleanup()
{
	//if(timeUntilBreak != NULL) {
	//
	//delete timeUntilBreak;
	//timeUntilBreak = NULL;
	//}
	//if(vehicleName != NULL) {
	//
	//delete vehicleName;
	//vehicleName = NULL;
	//}
	//if(drivingInViolationToday != NULL) {
	//
	//delete drivingInViolationToday;
	//drivingInViolationToday = NULL;
	//}
	//if(driverId != NULL) {
	//
	//delete driverId;
	//driverId = NULL;
	//}
	//if(cycleRemaining != NULL) {
	//
	//delete cycleRemaining;
	//cycleRemaining = NULL;
	//}
	//if(driverName != NULL) {
	//
	//delete driverName;
	//driverName = NULL;
	//}
	//if(dutyStatus != NULL) {
	//
	//delete dutyStatus;
	//dutyStatus = NULL;
	//}
	//if(cycleTomorrow != NULL) {
	//
	//delete cycleTomorrow;
	//cycleTomorrow = NULL;
	//}
	//if(shiftDriveRemaining != NULL) {
	//
	//delete shiftDriveRemaining;
	//shiftDriveRemaining = NULL;
	//}
	//if(timeInCurrentStatus != NULL) {
	//
	//delete timeInCurrentStatus;
	//timeInCurrentStatus = NULL;
	//}
	//if(drivingInViolationCycle != NULL) {
	//
	//delete drivingInViolationCycle;
	//drivingInViolationCycle = NULL;
	//}
	//if(shiftRemaining != NULL) {
	//
	//delete shiftRemaining;
	//shiftRemaining = NULL;
	//}
	//
}

void
HosLogsSummaryResponse_drivers::fromJson(char* jsonStr)
{
	JsonObject *pJsonObject = json_node_get_object(json_from_string(jsonStr,NULL));
	JsonNode *node;
	const gchar *timeUntilBreakKey = "timeUntilBreak";
	node = json_object_get_member(pJsonObject, timeUntilBreakKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&timeUntilBreak, node, "long long", "");
		} else {
			
		}
	}
	const gchar *vehicleNameKey = "vehicleName";
	node = json_object_get_member(pJsonObject, vehicleNameKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&vehicleName, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *drivingInViolationTodayKey = "drivingInViolationToday";
	node = json_object_get_member(pJsonObject, drivingInViolationTodayKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&drivingInViolationToday, node, "long long", "");
		} else {
			
		}
	}
	const gchar *driverIdKey = "driverId";
	node = json_object_get_member(pJsonObject, driverIdKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&driverId, node, "long long", "");
		} else {
			
		}
	}
	const gchar *cycleRemainingKey = "cycleRemaining";
	node = json_object_get_member(pJsonObject, cycleRemainingKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&cycleRemaining, node, "long long", "");
		} else {
			
		}
	}
	const gchar *driverNameKey = "driverName";
	node = json_object_get_member(pJsonObject, driverNameKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&driverName, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *dutyStatusKey = "dutyStatus";
	node = json_object_get_member(pJsonObject, dutyStatusKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&dutyStatus, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *cycleTomorrowKey = "cycleTomorrow";
	node = json_object_get_member(pJsonObject, cycleTomorrowKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&cycleTomorrow, node, "long long", "");
		} else {
			
		}
	}
	const gchar *shiftDriveRemainingKey = "shiftDriveRemaining";
	node = json_object_get_member(pJsonObject, shiftDriveRemainingKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&shiftDriveRemaining, node, "long long", "");
		} else {
			
		}
	}
	const gchar *timeInCurrentStatusKey = "timeInCurrentStatus";
	node = json_object_get_member(pJsonObject, timeInCurrentStatusKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&timeInCurrentStatus, node, "long long", "");
		} else {
			
		}
	}
	const gchar *drivingInViolationCycleKey = "drivingInViolationCycle";
	node = json_object_get_member(pJsonObject, drivingInViolationCycleKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&drivingInViolationCycle, node, "long long", "");
		} else {
			
		}
	}
	const gchar *shiftRemainingKey = "shiftRemaining";
	node = json_object_get_member(pJsonObject, shiftRemainingKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&shiftRemaining, node, "long long", "");
		} else {
			
		}
	}
}

HosLogsSummaryResponse_drivers::HosLogsSummaryResponse_drivers(char* json)
{
	this->fromJson(json);
}

char*
HosLogsSummaryResponse_drivers::toJson()
{
	JsonObject *pJsonObject = json_object_new();
	JsonNode *node;
	if (isprimitive("long long")) {
		long long obj = getTimeUntilBreak();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *timeUntilBreakKey = "timeUntilBreak";
	json_object_set_member(pJsonObject, timeUntilBreakKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getVehicleName();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *vehicleNameKey = "vehicleName";
	json_object_set_member(pJsonObject, vehicleNameKey, node);
	if (isprimitive("long long")) {
		long long obj = getDrivingInViolationToday();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *drivingInViolationTodayKey = "drivingInViolationToday";
	json_object_set_member(pJsonObject, drivingInViolationTodayKey, node);
	if (isprimitive("long long")) {
		long long obj = getDriverId();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *driverIdKey = "driverId";
	json_object_set_member(pJsonObject, driverIdKey, node);
	if (isprimitive("long long")) {
		long long obj = getCycleRemaining();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *cycleRemainingKey = "cycleRemaining";
	json_object_set_member(pJsonObject, cycleRemainingKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getDriverName();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *driverNameKey = "driverName";
	json_object_set_member(pJsonObject, driverNameKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getDutyStatus();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *dutyStatusKey = "dutyStatus";
	json_object_set_member(pJsonObject, dutyStatusKey, node);
	if (isprimitive("long long")) {
		long long obj = getCycleTomorrow();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *cycleTomorrowKey = "cycleTomorrow";
	json_object_set_member(pJsonObject, cycleTomorrowKey, node);
	if (isprimitive("long long")) {
		long long obj = getShiftDriveRemaining();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *shiftDriveRemainingKey = "shiftDriveRemaining";
	json_object_set_member(pJsonObject, shiftDriveRemainingKey, node);
	if (isprimitive("long long")) {
		long long obj = getTimeInCurrentStatus();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *timeInCurrentStatusKey = "timeInCurrentStatus";
	json_object_set_member(pJsonObject, timeInCurrentStatusKey, node);
	if (isprimitive("long long")) {
		long long obj = getDrivingInViolationCycle();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *drivingInViolationCycleKey = "drivingInViolationCycle";
	json_object_set_member(pJsonObject, drivingInViolationCycleKey, node);
	if (isprimitive("long long")) {
		long long obj = getShiftRemaining();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *shiftRemainingKey = "shiftRemaining";
	json_object_set_member(pJsonObject, shiftRemainingKey, node);
	node = json_node_alloc();
	json_node_init(node, JSON_NODE_OBJECT);
	json_node_take_object(node, pJsonObject);
	char * ret = json_to_string(node, false);
	json_node_free(node);
	return ret;
}

long long
HosLogsSummaryResponse_drivers::getTimeUntilBreak()
{
	return timeUntilBreak;
}

void
HosLogsSummaryResponse_drivers::setTimeUntilBreak(long long  timeUntilBreak)
{
	this->timeUntilBreak = timeUntilBreak;
}

std::string
HosLogsSummaryResponse_drivers::getVehicleName()
{
	return vehicleName;
}

void
HosLogsSummaryResponse_drivers::setVehicleName(std::string  vehicleName)
{
	this->vehicleName = vehicleName;
}

long long
HosLogsSummaryResponse_drivers::getDrivingInViolationToday()
{
	return drivingInViolationToday;
}

void
HosLogsSummaryResponse_drivers::setDrivingInViolationToday(long long  drivingInViolationToday)
{
	this->drivingInViolationToday = drivingInViolationToday;
}

long long
HosLogsSummaryResponse_drivers::getDriverId()
{
	return driverId;
}

void
HosLogsSummaryResponse_drivers::setDriverId(long long  driverId)
{
	this->driverId = driverId;
}

long long
HosLogsSummaryResponse_drivers::getCycleRemaining()
{
	return cycleRemaining;
}

void
HosLogsSummaryResponse_drivers::setCycleRemaining(long long  cycleRemaining)
{
	this->cycleRemaining = cycleRemaining;
}

std::string
HosLogsSummaryResponse_drivers::getDriverName()
{
	return driverName;
}

void
HosLogsSummaryResponse_drivers::setDriverName(std::string  driverName)
{
	this->driverName = driverName;
}

std::string
HosLogsSummaryResponse_drivers::getDutyStatus()
{
	return dutyStatus;
}

void
HosLogsSummaryResponse_drivers::setDutyStatus(std::string  dutyStatus)
{
	this->dutyStatus = dutyStatus;
}

long long
HosLogsSummaryResponse_drivers::getCycleTomorrow()
{
	return cycleTomorrow;
}

void
HosLogsSummaryResponse_drivers::setCycleTomorrow(long long  cycleTomorrow)
{
	this->cycleTomorrow = cycleTomorrow;
}

long long
HosLogsSummaryResponse_drivers::getShiftDriveRemaining()
{
	return shiftDriveRemaining;
}

void
HosLogsSummaryResponse_drivers::setShiftDriveRemaining(long long  shiftDriveRemaining)
{
	this->shiftDriveRemaining = shiftDriveRemaining;
}

long long
HosLogsSummaryResponse_drivers::getTimeInCurrentStatus()
{
	return timeInCurrentStatus;
}

void
HosLogsSummaryResponse_drivers::setTimeInCurrentStatus(long long  timeInCurrentStatus)
{
	this->timeInCurrentStatus = timeInCurrentStatus;
}

long long
HosLogsSummaryResponse_drivers::getDrivingInViolationCycle()
{
	return drivingInViolationCycle;
}

void
HosLogsSummaryResponse_drivers::setDrivingInViolationCycle(long long  drivingInViolationCycle)
{
	this->drivingInViolationCycle = drivingInViolationCycle;
}

long long
HosLogsSummaryResponse_drivers::getShiftRemaining()
{
	return shiftRemaining;
}

void
HosLogsSummaryResponse_drivers::setShiftRemaining(long long  shiftRemaining)
{
	this->shiftRemaining = shiftRemaining;
}

File: openapi-generator/java-vertx/src/main/java/org/openapitools/server/api/model/HosLogsSummaryResponseDrivers.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL) 
public class HosLogsSummaryResponseDrivers   {
  
  private Long timeUntilBreak;
  private String vehicleName;
  private Long drivingInViolationToday;
  private Long driverId;
  private Long cycleRemaining;
  private String driverName;
  private String dutyStatus;
  private Long cycleTomorrow;
  private Long shiftDriveRemaining;
  private Long timeInCurrentStatus;
  private Long drivingInViolationCycle;
  private Long shiftRemaining;

  public HosLogsSummaryResponseDrivers () {

  }

  public HosLogsSummaryResponseDrivers (Long timeUntilBreak, String vehicleName, Long drivingInViolationToday, Long driverId, Long cycleRemaining, String driverName, String dutyStatus, Long cycleTomorrow, Long shiftDriveRemaining, Long timeInCurrentStatus, Long drivingInViolationCycle, Long shiftRemaining) {
    this.timeUntilBreak = timeUntilBreak;
    this.vehicleName = vehicleName;
    this.drivingInViolationToday = drivingInViolationToday;
    this.driverId = driverId;
    this.cycleRemaining = cycleRemaining;
    this.driverName = driverName;
    this.dutyStatus = dutyStatus;
    this.cycleTomorrow = cycleTomorrow;
    this.shiftDriveRemaining = shiftDriveRemaining;
    this.timeInCurrentStatus = timeInCurrentStatus;
    this.drivingInViolationCycle = drivingInViolationCycle;
    this.shiftRemaining = shiftRemaining;
  }

    
  @JsonProperty("timeUntilBreak")
  public Long getTimeUntilBreak() {
    return timeUntilBreak;
  }
  public void setTimeUntilBreak(Long timeUntilBreak) {
    this.timeUntilBreak = timeUntilBreak;
  }

    
  @JsonProperty("vehicleName")
  public String getVehicleName() {
    return vehicleName;
  }
  public void setVehicleName(String vehicleName) {
    this.vehicleName = vehicleName;
  }

    
  @JsonProperty("drivingInViolationToday")
  public Long getDrivingInViolationToday() {
    return drivingInViolationToday;
  }
  public void setDrivingInViolationToday(Long drivingInViolationToday) {
    this.drivingInViolationToday = drivingInViolationToday;
  }

    
  @JsonProperty("driverId")
  public Long getDriverId() {
    return driverId;
  }
  public void setDriverId(Long driverId) {
    this.driverId = driverId;
  }

    
  @JsonProperty("cycleRemaining")
  public Long getCycleRemaining() {
    return cycleRemaining;
  }
  public void setCycleRemaining(Long cycleRemaining) {
    this.cycleRemaining = cycleRemaining;
  }

    
  @JsonProperty("driverName")
  public String getDriverName() {
    return driverName;
  }
  public void setDriverName(String driverName) {
    this.driverName = driverName;
  }

    
  @JsonProperty("dutyStatus")
  public String getDutyStatus() {
    return dutyStatus;
  }
  public void setDutyStatus(String dutyStatus) {
    this.dutyStatus = dutyStatus;
  }

    
  @JsonProperty("cycleTomorrow")
  public Long getCycleTomorrow() {
    return cycleTomorrow;
  }
  public void setCycleTomorrow(Long cycleTomorrow) {
    this.cycleTomorrow = cycleTomorrow;
  }

    
  @JsonProperty("shiftDriveRemaining")
  public Long getShiftDriveRemaining() {
    return shiftDriveRemaining;
  }
  public void setShiftDriveRemaining(Long shiftDriveRemaining) {
    this.shiftDriveRemaining = shiftDriveRemaining;
  }

    
  @JsonProperty("timeInCurrentStatus")
  public Long getTimeInCurrentStatus() {
    return timeInCurrentStatus;
  }
  public void setTimeInCurrentStatus(Long timeInCurrentStatus) {
    this.timeInCurrentStatus = timeInCurrentStatus;
  }

    
  @JsonProperty("drivingInViolationCycle")
  public Long getDrivingInViolationCycle() {
    return drivingInViolationCycle;
  }
  public void setDrivingInViolationCycle(Long drivingInViolationCycle) {
    this.drivingInViolationCycle = drivingInViolationCycle;
  }

    
  @JsonProperty("shiftRemaining")
  public Long getShiftRemaining() {
    return shiftRemaining;
  }
  public void setShiftRemaining(Long shiftRemaining) {
    this.shiftRemaining = shiftRemaining;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    HosLogsSummaryResponseDrivers hosLogsSummaryResponseDrivers = (HosLogsSummaryResponseDrivers) o;
    return Objects.equals(timeUntilBreak, hosLogsSummaryResponseDrivers.timeUntilBreak) &&
        Objects.equals(vehicleName, hosLogsSummaryResponseDrivers.vehicleName) &&
        Objects.equals(drivingInViolationToday, hosLogsSummaryResponseDrivers.drivingInViolationToday) &&
        Objects.equals(driverId, hosLogsSummaryResponseDrivers.driverId) &&
        Objects.equals(cycleRemaining, hosLogsSummaryResponseDrivers.cycleRemaining) &&
        Objects.equals(driverName, hosLogsSummaryResponseDrivers.driverName) &&
        Objects.equals(dutyStatus, hosLogsSummaryResponseDrivers.dutyStatus) &&
        Objects.equals(cycleTomorrow, hosLogsSummaryResponseDrivers.cycleTomorrow) &&
        Objects.equals(shiftDriveRemaining, hosLogsSummaryResponseDrivers.shiftDriveRemaining) &&
        Objects.equals(timeInCurrentStatus, hosLogsSummaryResponseDrivers.timeInCurrentStatus) &&
        Objects.equals(drivingInViolationCycle, hosLogsSummaryResponseDrivers.drivingInViolationCycle) &&
        Objects.equals(shiftRemaining, hosLogsSummaryResponseDrivers.shiftRemaining);
  }

  @Override
  public int hashCode() {
    return Objects.hash(timeUntilBreak, vehicleName, drivingInViolationToday, driverId, cycleRemaining, driverName, dutyStatus, cycleTomorrow, shiftDriveRemaining, timeInCurrentStatus, drivingInViolationCycle, shiftRemaining);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class HosLogsSummaryResponseDrivers {\n");
    
    sb.append("    timeUntilBreak: ").append(toIndentedString(timeUntilBreak)).append("\n");
    sb.append("    vehicleName: ").append(toIndentedString(vehicleName)).append("\n");
    sb.append("    drivingInViolationToday: ").append(toIndentedString(drivingInViolationToday)).append("\n");
    sb.append("    driverId: ").append(toIndentedString(driverId)).append("\n");
    sb.append("    cycleRemaining: ").append(toIndentedString(cycleRemaining)).append("\n");
    sb.append("    driverName: ").append(toIndentedString(driverName)).append("\n");
    sb.append("    dutyStatus: ").append(toIndentedString(dutyStatus)).append("\n");
    sb.append("    cycleTomorrow: ").append(toIndentedString(cycleTomorrow)).append("\n");
    sb.append("    shiftDriveRemaining: ").append(toIndentedString(shiftDriveRemaining)).append("\n");
    sb.append("    timeInCurrentStatus: ").append(toIndentedString(timeInCurrentStatus)).append("\n");
    sb.append("    drivingInViolationCycle: ").append(toIndentedString(drivingInViolationCycle)).append("\n");
    sb.append("    shiftRemaining: ").append(toIndentedString(shiftRemaining)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
